File: Prove/Python/01 - EsercitazioneOOP-Concorrenza/prod_cons_CV/package_coda/coda.py
from .ICoda import ICoda 
import logging
logging.basicConfig(level=logging.DEBUG, format='[%(threadName)-0s] %(message)s',)

class coda(ICoda):

    def __init__(self,dim):
        self.size = dim #QUEUE_SIZE
        self.codaCircolare = [None]*self.size  #Vettore vuoto         
        self.head = 0 #Testa coda per l'inserimento
        self.occ = 0 #numero di elementi
        self.tail = 0 #Tail coda per prelevare

    def getSize(self):
        return self.size

    def empty(self):
        return self.occ == 0

    def full(self):
        return self.occ == self.size

    def inserisci(self,elem):
        self.codaCircolare[self.head%self.size] = elem
        self.head = self.head+1
        self.occ = self.occ+1  
        logging.debug("Elemento inserito in coda.")
        

    def preleva(self):
        elem = self.codaCircolare[self.tail%self.size]
        self.occ+=-1
        self.tail+=1
        return elem

    

File: Prove/Java/06 - JMS/Esercitazioni/05_JSM_RMI/Esercitazione JMS-RMI_1/Giuleppe/src/coda/Coda.java
package coda;

public class Coda implements ICoda{

    private int[] coda;
    private int size;
    private int head;
    private int tail;
    private int occ;

    public Coda(int size){
        this.size = size;
        this.coda = new int[this.size];
        this.head = this.tail = this.occ=0;
    }

    @Override
    public void deposito(int val) {
        coda[head%size]=val;
        head++;
        occ++;
        System.out.println("[Coda] E' stato inserito il valore: "+val+".");
    }

    @Override
    public int preleva() {
        int val = coda[tail%size];
        tail++;
        occ--;

        System.out.println("[Coda] E' stato prelevato il valore: "+val);

        return val;
    }

    @Override
    public boolean empty() {
       return occ == 0;
    }

    @Override
    public boolean full() {
        return occ == size;
    }

    @Override
    public int getSize() {
        return this.size;
    }
    
}
